packages/material/src/complex/table-array.control.tsx

```diff
@@ -41,117 +55,325 @@ export const tableArrayTester: RankedTester = rankWith(10, and(
         && (schema.items as JsonSchema).type === 'object'
     ))
 );
+export interface TableState {
+  selected: any[];
+  openConfirmDelete: boolean;
+}
+export class TableArrayControl extends Renderer<TableProps, TableState> {
+  constructor(props) {
+    super(props);
 
-export class TableArrayControl extends Renderer<ControlProps, void> {
-
-  // TODO duplicate code
-  addNewItem(path: string) {
-    const element = {};
-    this.props.dispatch(
-      update(
-        path,
-        array => {
-          if (array === undefined || array === null) {
-            return [element];
-          }
-
-          const clone = _.clone(array);
-          clone.push(element);
-
-          return clone;
-        }
-      )
-    );
+    this.state = {
+      selected: [],
+      openConfirmDelete: false
+    };
   }
 
-  /**
-   * @inheritDoc
-   */
   render() {
-    const { uischema, schema, path, data, visible, errors, label } = this.props;
+    const { uischema, schema, path, data, visible, childErrors } = this.props;
     const controlElement = uischema as ControlElement;
-
     const tableClass = JsonForms.stylingRegistry.getAsClassName('array-table.table');
-    const labelClass = JsonForms.stylingRegistry.getAsClassName('array-table.label');
-    const buttonClass = JsonForms.stylingRegistry.getAsClassName('array-table.button');
     const controlClass = [JsonForms.stylingRegistry.getAsClassName('array-table'),
-      Helpers.convertToValidClassName(controlElement.scope.$ref)].join(' ');
+    Helpers.convertToValidClassName(controlElement.scope.$ref)].join(' ');
 
     const resolvedSchema = resolveSchema(schema, controlElement.scope.$ref + '/items');
-    const createControlElement = (key: string): ControlElement => ({
-      type: 'Control',
-      label: false,
-      scope: { $ref: `#/properties/${key}` }
-    });
-    const labelObject = Helpers.createLabelDescriptionFrom(controlElement);
-    const isValid = errors.length === 0;
-    const divClassNames = 'validation' + (isValid ? '' : ' validation_error');
+
+    const handleSelectAllClick = (_event, checked) => {
+      if (checked) {
+        this.setState({ selected: data });
+
+        return;
+      }
+      this.setState({ selected: [] });
+    };
+
+    const handleClick = (_event, child) => {
+      const selected = this.state.selected.filter(s => data.indexOf(s) !== -1);
+      const selectedIndex = selected.indexOf(child);
+      let newSelected = [];
```

Comment:
I'd remove the initialization and instead add an else case in the following if-else-clause. This seems clearer to since we perform the `concat` with an empty array anyway in all the cases so far, which seems unnecessary.